sql/informix-sql/InformixSQLLexer.g4

```diff
@@ -249,6 +250,21 @@ STRING_LITERAL: '\'' (~'\'' | '\'\'')* '\'';
 
 BLOB_LITERAL: 'X' STRING_LITERAL;
 
+NATIONAL_CHAR_STRING_LIT: 'N' '\'' (~('\'' | '\r' | '\n' ) | '\'' '\'' | NEWLINE)* '\'';
+
+CHAR_STRING: '\''  (~('\'' | '\r' | '\n') | '\'' '\'' | NEWLINE)* '\'';
+
+//use quoted string as char_string
+CHAR_STRING_PERL    : 'Q' '\'' (QS_ANGLE | QS_BRACE | QS_BRACK | QS_PAREN | QS_EXCLAM | QS_SHARP | QS_QUOTE | QS_DQUOTE) '\'' -> type(CHAR_STRING);
+fragment QS_ANGLE   : '<' .*? '>';
+fragment QS_BRACE   : '{' .*? '}';
+fragment QS_BRACK   : '[' .*? ']';
+fragment QS_PAREN   : '(' .*? ')';
+fragment QS_EXCLAM  : '!' .*? '!';
+fragment QS_SHARP   : '#' .*? '#';
+fragment QS_QUOTE   : '\'' .*? '\'';
+fragment QS_DQUOTE  : '"' .*? '"';
```

Comment:
Can we get rid of greedy syntax for these tokens? Greedyness could make understanding more difficult and could be more slowly.

```suggestion
fragment QS_ANGLE   : '<' ~'>'* '>';
fragment QS_BRACE   : '{' ~'}'* '}';
fragment QS_BRACK   : '[' ~']'* ']';
fragment QS_PAREN   : '(' ~')'* ')';
fragment QS_EXCLAM  : '!' ~'!'* '!';
fragment QS_SHARP   : '#' ~'#'* '#';
fragment QS_QUOTE   : '\'' ~'\''* '\'';
fragment QS_DQUOTE  : '"' ~'"'* '"';
```